reports/aksversions/main.py

```diff
@@ -0,0 +1,125 @@
+            resource_group_name = cluster.id.split('/')[4]
+
+            print("Found AKS cluster: " + cluster.name)
+            print()
+
+            # Get the upgrade profile for the cluster
+            upgrade_profile = container_client.managed_clusters.get_upgrade_profile(resource_group_name, cluster.name)
+
+            print("Checking AKS cluster: " + cluster.name + " for updates...")
+            
+            # Get the current version and available updates from the upgrade profile
+            current_version = upgrade_profile.control_plane_profile.kubernetes_version
+            updates = upgrade_profile.control_plane_profile.upgrades
+
+            if updates is None:
+                available_updates = "No updates available"
+                verdict = "No update required"
+                color_code = "green"
+            else:
+                latest = [upgrade.kubernetes_version for upgrade in updates]
+                for upgrade in updates:
+                    if upgrade.is_preview:
+                        available_updates = "Only Preview available"
+                        verdict = "Wait for general availability"
+                        color_code = "orange"
+                    elif ((get_minor_version(upgrade.kubernetes_version) - get_minor_version(current_version)) < 1 ):
+                        available_updates = upgrade.kubernetes_version
+                        verdict = "Patch version update only"
+                        color_code = "orange"
+                    else:
+                        available_updates = upgrade.kubernetes_version
+                        verdict = "Update"
+                        color_code = "red"
+
+            print("Creating output object for: " + cluster.name)
+            print()
+
+            # Create a dictionary with all the cluster information
+            cluster = {
+                "id": str(uuid.uuid4()),
+                "subscription": sub_details.display_name,
+                "clusterName": cluster.name,
+                "currentVersion": current_version,
+                "upgradeableVersion": max(available_updates) if type(available_updates) is list else available_updates,
+                "colorCode": color_code,
+                "resourceType": "CFT Cluster" if "cft" in cluster.name else "SS Cluster",
+                "powerState": cluster.power_state.code if cluster.power_state else "Unknown",
+                "verdict": verdict
+            }
+
+            # Add the dictionary to the list
+            clusters_info.append(cluster)
+
+# Print the list of all cluster information as a JSON object
+print(f"Number of AKS Clusters found: {len(clusters_info)}.")
+print()
+# print(json.dumps(clusters_info, indent=4))
+
+# Define the script and arguments
+script = "save-to-cosmos.py"
+args = [json.dumps(clusters_info)]
+
+# Try to run the script with arguments
+try:
+    result = subprocess.run(["python", script] + args, check=True)
```

Comment:
this doesn't seem right, why are we running as a subprocess rather than just calling a function?

Reply:
Updated to import functions and run all code in main file.